sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/EmptyMethod.CSharp9.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+void Empty() { } // FN
+
+void WithComment()
+{
+    // because
+}
+
+void NotEmpty()
+{
+    Console.WriteLine();
+}
+
+record EmptyMethod
+{
+    void F2()
+    {
+        // Do nothing because of X and Y.
+    }
+
+    void F3()
+    {
+        Console.WriteLine();
+    }
+
+    [Conditional("DEBUG")]
+    void F4()    // Noncompliant {{Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.}}
+    {
+    }
+
+    protected virtual void F5()
+    {
+    }
+
+    extern void F6();
+
+    [DllImport("avifil32.dll")]
+    private static extern void F7();
+}
+
+abstract record MyR
+{
+    void F1() { } // Noncompliant
+    public abstract void F2();
+}
+
+record MyR2 : MyR
+{
+    public override void F2()
+    {
+    }
+}
+
+class WithProp
+{
+    public string Prop
+    {
+        init { }
```

Comment:
Shouldn't this be a FN since it's not an auto-implemented property?

Reply:
the current rule only looks at methods, so it's the same behavior as for `set`.
I've opened https://github.com/SonarSource/sonar-dotnet/issues/3753 to do it

Reply:
Shouldn't we mark both as FNs then?